internal/cli/config/delete.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"fmt"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/mongodb/mongocli/internal/prompt"
+
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/description"
+	"github.com/spf13/cobra"
+)
+
+type DeleteOpts struct {
+	name string
+}
+
+func (opts *DeleteOpts) Run() error {
+	config.SetName(&opts.name)
+
+	profile := config.GetConfigDescription()
+	if len(profile) == 0 {
+		return fmt.Errorf("profile %v does not exist", opts.name)
+	}
+
+	shouldDelete := false
+	p := prompt.NewDeleteConfirm(opts.name)
+	if err := survey.AskOne(p, &shouldDelete); err != nil {
+		return err
+	}
```

Comment:
we have defined a pattern for this as we have a lot of commands that do this (delete and confirm) you can create a composition in your `DeleteOpts` of the `cli.DeleteOpts` and this should give you access to a Prompt() method you can use in the PreRun block of the `cobra.Command`

you can see an example [here](https://github.com/mongodb/mongocli/blob/238a113358ce6add54198173f430cc6d96c369e8/internal/cli/projects/delete.go)